library/dicom_frame.php

```diff
@@ -26,119 +29,471 @@
-    }
-
-    /* Safari */
-    @-webkit-keyframes spin {
-        0% {
-            -webkit-transform: rotate(0deg);
-        }
-        100% {
-            -webkit-transform: rotate(360deg);
-        }
-    }
-
-    @keyframes spin {
-        0% {
-            transform: rotate(0deg);
-        }
-        100% {
-            transform: rotate(360deg);
-        }
+        left: 15%;
+        visibility: hidden;
+        transform: scale(1.05);
+        transition: visibility 0s linear 0.25s, opacity 0.25s 0s, transform 0.25s;
+        z-index: 1000;
+    }
+
+    #loaderlist.popup {
+        top: 50%;
+        left: 50%;
+        z-index: 1010;
+    }
+
+    .loaderSelect {
+        display: inline;
+        width: 50%;
+    }
+
+    #loaderlist div.popup-content {
+        position: absolute;
+        top: 50%;
+        left: 50%;
+        transform: translate(-50%, -50%);
+        background-color: white;
+        padding: 1rem 1.25rem;
+        width: 24rem;
+        border-radius: 0.5rem;
+        padding-left: .325rem;
+    }
+
+    .lg div.popup-content {
+        position: static;
+        border: 2px solid #fff;
+        width: 75vw;
+        border-radius: 0.5rem;
+        overflow: auto;
+        height: 75vh;
+        background-color: white;
+    }
+
+    #helpPopup div.popup-content {
+        background-color: black;
+    }
+
+    .close-button {
+        background: #ff5d5a;
+        position: absolute;
+        line-height: 1.5rem;
+        text-align: center;
+        right: 20px;
+        width: 1.5rem;
+        height: 1.5rem;
+        cursor: pointer;
+        z-index: 1011;
+    }
+
+    .close-button:hover {
+        background-color: lightcoral;
+    }
+
+    .show-popup {
+        opacity: 1;
+        visibility: visible;
+        transform: scale(1.0);
+        transition: visibility 0s linear 0s, opacity 0.25s 0s, transform 0.25s;
+    }
+
+    #progressbar {
+        width: 0;
+        max-width: 92%;
+        height: 1.25rem;
+        background-color: #4CAF50;
+    }
+
+    #pageHeader h2 em {
+        font-size: 14px;
+        color: red;
+    }
+
+    div.toolList {
+        margin: .325rem 0 .325rem;
     }
 </style>
 <body>
-<!-- DWV -->
-<div id="dwv" src='<?php echo $web_path ?>'>
-    <!-- Toolbar -->
-    <div class="toolbar"></div>
-    <div class="warn_diagnostic"><?php echo xlt('Not For Diagnostic Use') ?>
-        <!-- Layer Container -->
-        <div class="layerContainer">
-            <span class="loader"></span>
-            <canvas id="dwvimg" class="imageLayer"><?php echo xlt('Only for HTML5 compatible browsers.') ?></canvas>
-        </div>
-        <!-- /layerContainer -->
+    <!-- DWV -->
+    <div id="dwv" class="container" src='<?php echo $web_path ?>'>
+        <input type="hidden" id="state_url" value=<?php echo $state_url ?> />
+        <input type="hidden" id="csrf" value=<?php echo $csrf ?> />
+        <input type="hidden" id="doc_id" value=<?php echo attr($docid) ?> />
```

Comment:
shouln't there be quotes around all the 3 value's above?

Reply:
Not totally sure Brady. I had a dickens of a time with this and landed here. These are already being past as strings. Still, i'll look more closely

Reply:
@tywrenn may strike us with lightning if we are not careful :fearful: 

![tenor (30)](https://user-images.githubusercontent.com/278968/87121359-75965e00-c237-11ea-97f1-54c382d74feb.gif)

Reply:
Ya have to remember this is being generated at the server and not hard coded so, html hard coded rules don't necessarily apply.